### ISSUE TITLE
Firefox download apparently intercepted too late

### ISSUE THREAD

**Issue by swiehr** [belaviyo/download-with#16]
I experience quite often that Firefox already created download files (e.g. download_file & download_file.part) which are then intercepted and forwarded to FDM. This leads to either conflict (when download folder is same) or unwanted files (when FDM download folder differs from Firefox download folder). Had to change my FDM policy to ignore downloads if file already exists for that reason.

**Comment by belaviyo** [comment]
True, I personally use the right-click context menu item whenever possible to prevent the browser download module from being called. Unfortunately, in Firefox, the earliest time an extension gets called when a new download is requested would be after the file creation. In Chrome, things are smoother. As far as I know, there is no fix for this. I'll keep this open for suggestions.

**Comment by swiehr** [comment]
Short note: For me this already improved a lot by moving from old FDM (3.9.7) to FDM 5 (although I am missing some features in the latter, but that's a different story)

**Comment by Salvora** [comment]
Same issue here.
FF creates small files in default download location. I use Jdownloader and it uses different download location so there is no conflict but it is annoying to clean the firefox default download location folder every now and then.

**Comment by hunkyburrito** [comment]
I realize this is pretty late but from what I tested (with a modified jdownloader2 firefox add-on) the way to fix this is to change `chrome.downloads.pause(...)` to `chrome.downloads.cancel(...)` [here (or in applicable downloader code)](https://github.com/belaviyo/download-with/blob/725ded3ca061c189ae4e5fd7a10a97ddc0c8ddb1/common/common.js#L197). This way when the download is first started it will cancel/delete the file instead of just pausing and leaving leftovers.
